pass/disk.go

```diff
@@ -14,100 +14,126 @@ import (
 	sfuzzy "github.com/sahilm/fuzzy"
 )
 
+// StoreDefinition defines a password store object
+type StoreDefinition struct {
+	Name string `json:"name"`
+	Path string `json:"path"`
+}
+
 type diskStore struct {
-	path     string
+	stores   []StoreDefinition
 	useFuzzy bool // Setting for FuzzySearch or GlobSearch in manual searches
 }
 
-func NewDefaultStore() (Store, error) {
-	path, err := defaultStorePath()
-	if err != nil {
-		return nil, err
+func NewDefaultStore(stores []StoreDefinition, useFuzzy bool) (Store, error) {
+	if stores == nil || len(stores) == 0 {
+		defaultPath, err := defaultStorePath()
+		if err != nil {
+			return nil, err
+		}
+		stores = []StoreDefinition{{Name: "default", Path: defaultPath}}
 	}
 
-	return &diskStore{path, false}, nil
+	// Expand paths, follow symlinks
+	for i, store := range stores {
+		path := store.Path
+		if strings.HasPrefix(path, "~/") {
+			path = filepath.Join("$HOME", path[2:])
+		}
+		path = os.ExpandEnv(path)
+		path, err := filepath.EvalSymlinks(path)
+		if err != nil {
+			return nil, err
+		}
+		stores[i].Path = path
+	}
+
+	return &diskStore{stores, useFuzzy}, nil
 }
 
 func defaultStorePath() (string, error) {
-	usr, err := user.Current()
+	path := os.Getenv("PASSWORD_STORE_DIR")
+	if path != "" {
+		return path, nil
+	}
 
+	usr, err := user.Current()
 	if err != nil {
 		return "", err
 	}
 
-	path := os.Getenv("PASSWORD_STORE_DIR")
-	if path == "" {
-		path = filepath.Join(usr.HomeDir, ".password-store")
-	}
-
-	// Follow symlinks
-	return filepath.EvalSymlinks(path)
-}
-
-// Set the configuration options for password matching.
-func (s *diskStore) SetConfig(use_fuzzy *bool) error {
-	if use_fuzzy != nil {
-		s.useFuzzy = *use_fuzzy
-	}
-	return nil
+	path = filepath.Join(usr.HomeDir, ".password-store")
+	return path, nil
 }
 
 // Do a search. Will call into the correct algoritm (glob or fuzzy)
 // based on the settings present in the diskStore struct
 func (s *diskStore) Search(query string) ([]string, error) {
-	// default glob search
-	if !s.useFuzzy {
-		return s.GlobSearch(query)
-	} else {
+	if s.useFuzzy {
 		return s.FuzzySearch(query)
 	}
+	return s.GlobSearch(query)
 }
 
 // Fuzzy searches first get a list of all pass entries by doing a glob search
 // for the empty string, then apply appropriate logic to convert results to
 // a slice of strings, finally returning all of the unique entries.
 func (s *diskStore) FuzzySearch(query string) ([]string, error) {
-	var items []string
-	file_list, err := s.GlobSearch("")
+	entries, err := s.GlobSearch("")
 	if err != nil {
 		return nil, err
 	}
 
+	// GlobSearch now results `storename:filename`, for fuzzy search we need to provide only file names
+	var fileNames []string
+	for _, entry := range entries {
+		fileNames = append(fileNames, strings.SplitN(entry, ":", 2)[1])
+	}
```

Comment:
@erayd you'll love this 🙂 Caught by a unit test, that suddenly started finding 10 entries instead of 1 for a search term `"def"` (which started to match store name "default")

Reply:
Nice catch :-).

I actually use the store name as part of my search fairly commonly, to distinguish between files in different stores that have otherwise identical names. However, I don't actually need that capability in the host app - having it in the "refine search" filter (which also includes the store name in searches) is enough for my use-case :-).

Reply:
Good point, until now I didn't even think to test how "refine search" behaves with store names, but I agree this is a very common use-case to refine by the store name. In fact, I'm used to that too, because I have both "work" and "personal" in the same password store, but now I can configure different browser profiles to use different subfolders and this reduces 99% of cases where I needed to refine - now I just press Enter 🙂